x/mongo/driver/topology/pool_test.go

```diff
@@ -518,6 +520,34 @@ func TestPool(t *testing.T) {
 			}
 			close(cleanup)
 		})
+		t.Run("handshaker i/o fails", func(t *testing.T) {
```

Comment:
Just a question here... does this deadlock without the change?

Reply:
yup